### PR TITLE
Use also invisible tabs to calculate minimum size of tab container

### LINKED ISSUE #21404: TabContainer children can overflow on tab switch (e.g. Export and Project dialogs)
**Godot version:**
3.1.dev-1

**OS/device including version:**
Arch Linux

**Issue description:**
Export manager has too long textboxes: https://i.imgur.com/HvjGhzF.png

**Steps to reproduce:**
Open export window and enter resources menu.

_Bugsquad edit:_ 
hvjghzf

### LINKED ISSUE #25281: Empty scene with "Create Root Node" widget adds extra margin to SceneTree
**Godot version:**
Current master (7b4e7d56)

**OS/device including version:**
Mageia 7 Linux x86_64

**Issue description:**
When switching from a saved scene (with nodes in the SceneTree) to an empty new scene with the Create Root Node widget, the width of the SceneTree dock changes. The Create Root Node widget seems to add some extra margin.

It doesn't seem to be dependent on the existing width of the dock, even if I make it much wider than needed to show its contents, the empty scene will expand it.

screenshot_20190124_102928
screenshot_20190124_103407

Tested with French locale, but the same happens in English.

**Steps to reproduce:**
- Create a scene, add a root node, save it
- Create a scene, don't add a root node
- Switch between both scenes and see the SceneTree dock expand and contract

### LINKED ISSUE #25317: Inspector dock changes size when selected
**Godot version:**
 5b5db08

**Issue description:**
When clicking the Inspector dock, it will change its size. It only affects this dock alone, no matter its current location:
peek 2019-01-25 11-00

### LINKED ISSUE #25346: Tab container items overflow while resizing and then switching to another tab
**OS/device including version:**
Xubuntu 18.04

**Issue description:**
Tab container items overflow while resizing and then switching to another tab. It happens on the project settings window.

**Steps to reproduce:**
1) Open project settings window and switch to some other tab say AutoLoad
2) Resize the window to a smaller version
3) Now go the General tab

All items of the General tab will be overflowing
screenshot from 2019-01-26 21-15-12

### PR DESCRIPTION
**Pull request by azagaya**
Fixes #25281, fixes #25317, fixes #21404 and fixes #25346

### CODE REVIEW

**Comment by ankitpriyarup** [comment]
Works like a charm for me.

**Review by ScepticDope — APPROVED** [review]
Nice! :+1:

**Comment by akien-mga** [comment]
I'm not convinced this is right solution, as it's a feature currently that the invisible tabs are not taken into account to calculate the minimum size, and they should only be taken into account to resize everything when they are actually displayed.

I'm not sure what would be the better solution though, so keeping this open for discussion.

**Comment by azagaya** [comment]
> I'm not convinced this is right solution..

I totally agree with you. I've even tried to find another solution a long time after making this pull request without any luck. Mainly because there is no reason why the dock would resize, as all contents fits, and if it does, it shouldnt shrink back after that, cause minimum size would be set to the other size.
However, this is the only way i could think of solving also #25346.
Also, using the size of the bigger tab of the dock as minimum size (so it does not shrink when selecting other tab afterwards) imo is like using invisible tabs.

Also, for gui elements in my software i usually use Qt, and if you set a minimum size to an element in a tab, it affects the size of the other Tabs, so i thought that if a mature framework for gui like Qt did this, it wasn't so bad idea. (I don't mean that we should follow other soft design decisions, i'm just giving an example here). 

Anyways, i agree that i'm not fully convinced either, and not to use invisible tabs for size is a cool feature. This is more like a workarround, and i couldn't find any other solution.

**Comment by azagaya** [comment]
Another thing i've noticed is that changning some text in buttons inside tabs also appears to workarround this. With this i mean that possibly the bug comes from the size reported by the elements to the tab container, or something like that. I've tried in many languages and changing text made this symptom dissapear (obviously we cant fix this changing text, but may be it leads us to the root of the problem)

**Comment by groud** [comment]
We discussed what to do on IRC with that. We are ok on the feature provided it is made optional.
So basically it would require to add a boolean property, called somehow like "Use hidden tabs to compute min size", that would activate this behavior. The default should be disabled for compatibility reasons.

File: scene/gui/tab_container.cpp [hunk 1]
```diff
@@ -876,6 +886,7 @@ void TabContainer::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::INT, "current_tab", PROPERTY_HINT_RANGE, "-1,4096,1", PROPERTY_USAGE_EDITOR), "set_current_tab", "get_current_tab");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "tabs_visible"), "set_tabs_visible", "are_tabs_visible");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "drag_to_rearrange_enabled"), "set_drag_to_rearrange_enabled", "get_drag_to_rearrange_enabled");
+  ADD_PROPERTY(PropertyInfo(Variant::BOOL, "use_hidden_tabs_for_min_size"), "set_use_hidden_tabs_for_min_size", "get_use_hidden_tabs_for_min_size");
```
Comment (akien-mga) [inline scene/gui/tab_container.cpp]: Indentation issue. (Also further down and in header.)

Comment (groud) [reply scene/gui/tab_container.cpp]: The fix is done but in the wrong commit, in that case it might just be better to squash them I guess.

Comment (akien-mga) [reply scene/gui/tab_container.cpp]: Ah indeed.

Comment (azagaya) [reply scene/gui/tab_container.cpp]: Oh, you are right.. i fixed indentation in wrong commit. Sorry for that. So, Should i squash them?

Comment (akien-mga) [reply scene/gui/tab_container.cpp]: Yeah squashing them should be fine.

Comment (azagaya) [reply scene/gui/tab_container.cpp]: Done! sorry about that.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**